docs/06-concepts/05-sessions.md

```diff
@@ -1,26 +1,476 @@
+}
+```
+
+**Solution 2 - Create manual session:**
+
+```dart
+Future<void> processUser(Session session, int userId) async {
+  var user = await User.db.findById(session, userId);
+
+  Timer(Duration(seconds: 5), () async {
+    // Create new session for async work
+    var newSession = await Serverpod.instance.createSession();
+    try {
+      await user.updateLastSeen(newSession);
+    } finally {
+      await newSession.close();
+    }
+  });
+
+  return;
+}
+```
+
+### Pitfall 2: Forgetting to close manual sessions
+
+**Problem:**
+
+```dart
+// ❌ Memory leak!
+var session = await Serverpod.instance.createSession();
+var users = await User.db.find(session);
+// Forgot to close - session leaks memory
+```
+
+**Solution - Always use try-finally:**
+
+```dart
+var session = await Serverpod.instance.createSession();
+try {
+  var users = await User.db.find(session);
+  // Process users
+} finally {
+  await session.close(); // Always runs
+}
+```
+
+### Pitfall 3: Accessing global cache without Redis
+
+**Problem:**
+
+```dart
+// ❌ Throws assertion error if Redis not enabled
+await session.caches.global.put('key', data);
+```
+
+**Solution - Enable Redis in config:**
+
+```yaml
+# config/development.yaml
+redis:
+  enabled: true # Default is false
+  host: localhost
+  port: 6379
+```
+
+Or check before using:
+
+```dart
+// Use local cache as fallback
+if (Features.enableRedis) {
+  await session.caches.global.put('key', data);
+} else {
+  await session.caches.local.put('key', data);
+}
+```
+
+## Working with different session types
+
+Understanding when each session type appears in your code:
+
+### MethodCallSession - Standard API calls
+
+Created for: `Future<T>` endpoint methods
+Lifetime: Single request-response cycle
+Common in: User authentication, data fetching, CRUD operations
+
+```dart
+Future<String> hello(Session session, String name) async {
+  // You're working with a MethodCallSession here
+  // It will close automatically after this method returns
+  return 'Hello $name';
+}
+```
+
+### MethodStreamSession - Real-time communication
+
+Created for: `Stream<T>` endpoint methods or stream parameters
+Lifetime: Duration of the stream
+Common in: Chat applications, live notifications, real-time collaboration
+
+```dart
+Stream<Message> chat(Session session, Stream<String> incoming) async* {
+  // MethodStreamSession stays open for the stream duration
+  await for (var message in incoming) {
+    yield Message(text: message, time: DateTime.now());
+  }
+  // Closes when stream completes
+}
+```
+
+### FutureCallSession - Background tasks
+
+Created for: Scheduled operations
+Lifetime: Task execution duration
+Common in: Email sending, report generation, data cleanup
+
+```dart
+class EmailSender extends FutureCall {
+  @override
+  Future<void> invoke(Session session, EmailData? data) async {
+    // FutureCallSession created just for this task
+    await sendEmail(session, data!);
+    // Closes when task completes
+  }
+}
+```
```

Comment:
Comment: Not sure about this section. The problem I see is that it is not clear to me what is special about the session in each case, we are just using the normal Session interface. Which I btw think is good, but that kind of makes the value of this text questionable. 

No actionable feedback, I'm not sure what a better alternative is..